### PR TITLE
Migrating from buddybuild to bitrise

### PR DESCRIPTION
**Pull request by marianeum**

### CODE REVIEW

File: _posts/2018-01-15-migrating-from-buddybuild-to-bitrise.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+Bitrise is different. It has a lot of integrations which you can chain together as individual steps to a workflow. Each integration has its own configurations. If there's an integration missing you can add a custom script step with whatever logic you want. This gives you a lot of flexibility and control, but you still get simplicity of having pre-built integrations. All integrations are versioned, so an update to an integration should not break your builds.
+
+## How to set up your configuration?
+
+Your Bitrise configuration is a combination of triggers, workflows and steps. Steps are individual integrations that can be chained together in a workflow. Triggers (as the name suggests) define when a workflow will run. You can create triggers for pull requests, pushes and tags, or set up scheduled builds. For our setup we have triggers for pull requests, pushes on master, creating release tags and the scheduled nightly builds. Each of those triggers run different workflows to ensure the correct steps run.
+
+<figure class="small">
```
Comment (marianeum) [inline _posts/2018-01-15-migrating-from-buddybuild-to-bitrise.md]: ideally this would wrap with the text @kapowaz any easy way to do that?

File: _posts/2018-01-15-migrating-from-buddybuild-to-bitrise.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+We use CI for various things:
+- Run our unit and instrumentation tests to ensure there's no regressions
+- Create an APK and deploy it to fabric and/or Google Play
+- Create nightly builds to deploy to the Google Play alpha channel for internal testing
+
+We looked at various options, including Nevercode, Visual Studio App Centre and Circle CI. Bitrise turned out to be the most flexible and feature rich option out of those that could support all the different use cases we had.
+
+## What's the difference between Buddybuild and Bitrise?
+
+Buddybuild is set up to be very user friendly - changing your build settings is often just a simple toggle away. This means it's very easy to set up and change settings, but it also comes with a disadvantage: it's not very flexible. If the UI doesn't support something you had to either find a workaround or hope they would add support for it.
```
Comment (jphastings) [inline _posts/2018-01-15-migrating-from-buddybuild-to-bitrise.md]: "hope you would" should probably be "hope they will" :)

File: _posts/2018-01-15-migrating-from-buddybuild-to-bitrise.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+---
+
+## Why Bitrise?
+
+We use CI for various things:
+- Run our unit and instrumentation tests to ensure there's no regressions
+- Create an APK and deploy it to fabric and/or Google Play
+- Create nightly builds to deploy to the Google Play alpha channel for internal testing
+
+We looked at various options, including Nevercode, Visual Studio App Centre and Circle CI. Bitrise turned out to be the most flexible and feature rich option out of those that could support all the different use cases we had.
```
Comment (bencord0) [inline _posts/2018-01-15-migrating-from-buddybuild-to-bitrise.md]: `support all the different use cases we had`

Use the active voice?
`support all of our use cases`

File: _posts/2018-01-15-migrating-from-buddybuild-to-bitrise.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+Open the last tab of the workflow page called `bitrise.yml`. From here find the step you want to modify and add a "run_if" line. To check if our environment variable we had to use this code: {% raw %}`run_if: '{{enveq "PR_READY_FOR_QA" "true"}}'`{% endraw %}. That's it! Now the step will only run if the PR is ready for QA.
+
+<figure class="small">
+![Example of a run if condition](/images/posts/migrating-from-buddybuild-to-bitrise/runif-condition.jpg)
+</figure>
+
+## What's missing?
+
+There's a few things missing. One being a nice output for unit tests, but this is being worked on. Bitrise also doesn't have an SDK (yet?), so crash reporting and feedback won't be available. This isn't a big problem for us, but if it is for you you might want to look at other tools supporting feedback or crash reporting.
```
Comment (bencord0) [inline _posts/2018-01-15-migrating-from-buddybuild-to-bitrise.md]: `There are a few things missing`

File: _posts/2018-01-15-migrating-from-buddybuild-to-bitrise.md (hunk 1 above)
Comment (kapowaz) [reply _posts/2018-01-15-migrating-from-buddybuild-to-bitrise.md]: Not really with the current design, I’m afraid — it’s been built on the assumption that embedded images like this would exist in the main flow. You could wrap the `<figure>` in an `<aside>` block, and give it a heading/description like so:


```markdown
<aside>
#### blah blah

some description here

  <figure class="small">
  ![Part of our master workflow](/images/posts/migrating-from-buddybuild-to-bitrise/workflow.jpg)
  </figure>
</aside>
```

…but this doesn’t have the best layout; you end up with something like this:

img

If you’d like to do that and create an issue, I can have a go at improving the layout of figures inside aside blocks separately?

Comment (marianeum) [reply _posts/2018-01-15-migrating-from-buddybuild-to-bitrise.md]: ah that's not too bad, thanks!

File: _posts/2018-01-15-migrating-from-buddybuild-to-bitrise.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
+---
+layout: post
+title:  "Migrating from Buddybuild to Bitrise"
+authors:
+  - "Maria Neumayer"
+excerpt: >
+  At the start of the year Apple announced the acquisition of Buddybuild and stopping Android support by March. This meant we quickly had to find an alternative - Bitrise came out to be the best option. Here's why, and how we switched.
```
Comment (kapowaz) [inline _posts/2018-01-15-migrating-from-buddybuild-to-bitrise.md]: _‘stopping Android support’_ reads a little strangely to me. I’d suggest:

> At the start of the year Apple announced their acquisition of Buddybuild, along with the news that Android support would end by March.

Comment (kapowaz) [reply _posts/2018-01-15-migrating-from-buddybuild-to-bitrise.md]: Slight rewording suggestion:

> This meant we had to find an alternative quickly, and Bitrise was our eventual choice. Here's why, and how we switched.

File: _posts/2018-01-15-migrating-from-buddybuild-to-bitrise.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+---
+
+## Why Bitrise?
+
+We use CI for various things:
+- Run our unit and instrumentation tests to ensure there's no regressions
```
Comment (kapowaz) [inline _posts/2018-01-15-migrating-from-buddybuild-to-bitrise.md]: > Run our unit and instrumentation tests to ensure there are no regressions

File: _posts/2018-01-15-migrating-from-buddybuild-to-bitrise.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+- Create an APK and deploy it to fabric and/or Google Play
```
Comment (kapowaz) [inline _posts/2018-01-15-migrating-from-buddybuild-to-bitrise.md]: Titlecase _Fabric_, perhaps?

File: _posts/2018-01-15-migrating-from-buddybuild-to-bitrise.md (hunk 2 above)
Comment (kapowaz) [inline _posts/2018-01-15-migrating-from-buddybuild-to-bitrise.md]: > This means it's very easy to set up and change settings, but it also comes with a disadvantage: it's not very flexible.

It’s not clear to me why being easy to change settings means it’s not very flexible — might it be worth expanding on this? Or if they’re not related, maybe reword slightly (as it suggests that this ease of use is perhaps the cause?)

File: _posts/2018-01-15-migrating-from-buddybuild-to-bitrise.md (hunk 4 above)
Comment (kapowaz) [inline _posts/2018-01-15-migrating-from-buddybuild-to-bitrise.md]: Suggested edit:

> There are a few things missing: better output for unit tests is one, but this is being worked on.

File: _posts/2018-01-15-migrating-from-buddybuild-to-bitrise.md (hunk 2 above)
Comment (marianeum) [reply _posts/2018-01-15-migrating-from-buddybuild-to-bitrise.md]: ah yes basically because you can't change the actual task that is being run - you have to rely on them exposing the right toggles etc

File: _posts/2018-01-15-migrating-from-buddybuild-to-bitrise.md (hunk 5 above)
Comment (K4KYA) [reply _posts/2018-01-15-migrating-from-buddybuild-to-bitrise.md]: Another alternative:
> ...acquisition of Buddybuild, followed by the inevitable announcement of dropping Android support from March.

File: _posts/2018-01-15-migrating-from-buddybuild-to-bitrise.md (hunk 6 above)
Comment (K4KYA) [reply _posts/2018-01-15-migrating-from-buddybuild-to-bitrise.md]: > To run...

File: _posts/2018-01-15-migrating-from-buddybuild-to-bitrise.md [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+- Create nightly builds to deploy to the Google Play alpha channel for internal testing
```
Comment (K4KYA) [inline _posts/2018-01-15-migrating-from-buddybuild-to-bitrise.md]: > To deploy our internal nightly build on the Google Play alpha channel for internal feedback.

File: _posts/2018-01-15-migrating-from-buddybuild-to-bitrise.md (hunk 5 above)
Comment (marianeum) [reply _posts/2018-01-15-migrating-from-buddybuild-to-bitrise.md]: as much as i'd like to do some apple bashing not sure we should

File: _posts/2018-01-15-migrating-from-buddybuild-to-bitrise.md [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+
+Open the last tab of the workflow page called `bitrise.yml`. From here find the step you want to modify and add a "run_if" line. To check if our environment variable we had to use this code: {% raw %}`run_if: '{{enveq "PR_READY_FOR_QA" "true"}}'`{% endraw %}. That's it! Now the step will only run if the PR is ready for QA.
+
+<figure class="small">
+![Example of a run if condition](/images/posts/migrating-from-buddybuild-to-bitrise/runif-condition.jpg)
+</figure>
+
+## What's missing?
+
+There are a few things missing. One being a nice output for unit tests, but this is being worked on. Bitrise also doesn't have an SDK (yet?), so crash reporting and feedback won't be available. This isn't a big problem for us, but if it is for you you might want to look at other tools supporting feedback or crash reporting.
```
Comment (K4KYA) [inline _posts/2018-01-15-migrating-from-buddybuild-to-bitrise.md]: Maybe we can find an alternative to the double 'you'. Something like:
> Whilst this wasn't an issue for us, you may want to find an alternative if feedback and crash reporting are a hard requirement of your CI service.

File: _posts/2018-01-15-migrating-from-buddybuild-to-bitrise.md [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+---
+
+## Why Bitrise?
+
+We use CI for various things:
+- Run our unit and instrumentation tests to ensure there's no regressions
+- Create an APK and deploy it to fabric and/or Google Play
+- Create nightly builds to deploy to the Google Play alpha channel for internal testing
+
+We looked at various options, including Nevercode, Visual Studio App Centre and Circle CI. Bitrise turned out to be the most flexible and feature rich option out of those that could support all of our use cases.
```
Comment (bencord0) [inline _posts/2018-01-15-migrating-from-buddybuild-to-bitrise.md]: CircleCI have a blog post for people doing this: https://circleci.com/blog/migrating-from-buddybuild-to-circleci/

Maybe add a link to their post, do the others have corresponding docs?
